src/components/structures/auth/Registration.tsx

```diff
@@ -288,6 +296,7 @@ export default class Registration extends React.Component<IProps, IState> {
     };
 
     private onUIAuthFinished = async (success: boolean, response: any) => {
```

Comment:
Can we fix this `any`? It seems like it caused a bug prior to this change where it was checking `userId`

Reply:
Sure, I’ll try to figure out the real type and update it :)

Reply:
Yeah, doesn’t look like it’ll really be possible to do this properly.

This entire part of the JS SDK is still missing types and the types used here aren’t even close to what they should be (just one example: https://github.com/matrix-org/matrix-js-sdk/blob/develop/src/client.ts#L6840-L6849)

Reply:
Can we at least type the fields we use in this method and label it as partial? The `room_id` vs `roomId` is pretty disastrous.

Reply:
I could, but that’d still first require a change in matrix-js-sdk (we’d need to add the required fields to the AuthData interface), which would definitely be a T-Enhancement, so it’d require PM approval.

I’ve created branches with the minimal changes that’d be required for this:

https://github.com/matrix-org/matrix-js-sdk/commit/d7fa76a34afb796ebce97301b9ee4d5c8f3cc9ac
https://github.com/matrix-org/matrix-react-sdk/commit/3c2e1f5d1b0926454953209b601d77d8810da8db

Reply:
Adding types is `T-Task` - not `T-Enhancement` and thus does not require anything beyond code review.